RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/TechnicianService.java

```diff
@@ -277,7 +278,66 @@ public static TechnicianDto technicianToDTO(Technician tech) {
 		
 	}
 	
+	@Transactional // TODO check if this is what is needed for this method: this method supposed to remove absolutely all the timeslots and appointments from technicians shedule
+	public String deleteFullTechnicianWorkSchedule(String email) throws Exception{
+		
+		if(email == null) {
+			throw new Exception("Email cannot be empty.");
+		}
+
+		Technician technician = technicianRepository.findTechnicianByEmail(email);
+		
+		if(technician == null) {
+			throw new Exception("Technician not found.");
+		}
+		
+		// removing all the appointments
+		for (Appointment appointmentToRemove: technician.getAppointments()) {
+			technician.getAppointments().remove(appointmentToRemove);
+		}
+		// removing all the timeslots
+		for (TimeSlot timeSlotToRemove: technician.getTimeslots()) {
+			technician.getTimeslots().remove(timeSlotToRemove);
+		}
```

Comment:
I'm not sure, but I think we'll need to:  (_I need a confirmation_)

- [x] Add `@AutowiredTimeSlotRepository timeSlotRepository;` to this class 
- [ ] Add `	timeSlotRepository.save(timeSlot);  ` in `addTechnicianWorkHours()`
- [ ] Delete all the appointments & timeslots by id (See suggested change)
![image](https://user-images.githubusercontent.com/70775435/111160897-ed5c1b80-8570-11eb-9abe-5cccd12707f5.png)



```suggestion
		// removing all the appointments
		for (Appointment appointmentToRemove: technician.getAppointments()) {
			appRepo.deleteById(appointmentToRemove.getAppointmentID());  //delete it from AppointmentRepository
			technician.getAppointments().remove(appointmentToRemove);
		}
		// removing all the timeslots
		for (TimeSlot timeSlotToRemove: technician.getTimeslots()) {
			timeSlotRepository.deleteById(timeSlotToRemove.getTimeSlotID()); //delete it from TimeSlot Repository
			technician.getTimeslots().remove(timeSlotToRemove);
		}
```